docs/content/concepts/partitions-schedules-sensors/sensors.mdx

```diff
@@ -40,63 +40,48 @@ The [Dagster daemon](/deployment/dagster-daemon) runs each sensor evaluation fun
-    filename: str
+my_job = define_asset_job("my_job", [my_asset])
 
 
-@op
-def process_file(context: OpExecutionContext, config: FileConfig):
-    context.log.info(config.filename)
+@sensor(job=my_job)
+def my_asset_input_file_sensor():
+    last_modified_time = os.path.getmtime("inputfile.csv")
+    return RunRequest(run_key=str(last_modified_time))
 
 
-@job
-def log_file_job():
-    process_file()
+defs = Definitions(sensors=[my_asset_input_file_sensor], assets=[my_asset])
 ```
 
-You can write a sensor that watches for new files in a specific directory and yields a `RunRequest` for each new file in the directory. By default, this sensor runs every 30 seconds.
+You can write a sensor that watches for changes to the input file and requests a run to re-materialize `my_asset` each time it encounters one:
 
-```python file=concepts/partitions_schedules_sensors/sensors/sensors.py startafter=start_directory_sensor_marker endbefore=end_directory_sensor_marker
-import os
-from dagster import sensor, RunRequest, RunConfig
+```python file=concepts/partitions_schedules_sensors/sensors/sensors.py startafter=start_asset_job_sensor_marker endbefore=end_asset_job_sensor_marker
+from dagster import define_asset_job, sensor, RunRequest, Definitions
 
 
-@sensor(job=log_file_job)
-def my_directory_sensor():
-    for filename in os.listdir(MY_DIRECTORY):
-        filepath = os.path.join(MY_DIRECTORY, filename)
-        if os.path.isfile(filepath):
-            yield RunRequest(
-                run_key=filename,
-                run_config=RunConfig(
-                    ops={"process_file": FileConfig(filename=filename)}
-                ),
-            )
-```
+my_job = define_asset_job("my_job", [my_asset])
 
-This sensor iterates through all the files in `MY_DIRECTORY` and yields a <PyObject object="RunRequest"/> for each file. Note that despite the `yield` syntax, the function will run to completion before any runs are submitted.
 
-To write a sensor that materializes assets, you can [build a job that materializes assets](/concepts/assets/software-defined-assets#building-jobs-that-materialize-assets):
-
-```python file=concepts/partitions_schedules_sensors/sensors/sensors.py startafter=start_asset_job_sensor_marker endbefore=end_asset_job_sensor_marker
-asset_job = define_asset_job("asset_job", "*")
+@sensor(job=my_job)
+def my_asset_input_file_sensor():
+    last_modified_time = os.path.getmtime("inputfile.csv")
+    return RunRequest(run_key=str(last_modified_time))
 
 
-@sensor(job=asset_job)
-def materializes_asset_sensor():
-    yield RunRequest(...)
+defs = Definitions(sensors=[my_asset_input_file_sensor], assets=[my_asset])
 ```
 
+By default, this sensor runs every 30 seconds. It checks the last modified time for "inputfile.csv" and returns <PyObject object="RunRequest"/> based on it. Only one run will be launched for each unique value passed to the `run_key` parameter, which means that only one run will be launched each time the file is modified.
```

Comment:
```suggestion
By default, this sensor runs every 30 seconds. It checks the last modified time for `inputfile.csv` and returns <PyObject object="RunRequest"/> based on it. Only one run will be launched for each unique value passed to the `run_key` parameter, which means that only one run will be launched each time the file is modified.
```

Added some formatting, but also:

> checks the last modified time for `inputfile.csv` and returns <PyObject object="RunRequest"/> based on it

"Based on it" meaning what, exactly? That the file has a more recently modified timestamp than the last time it was checked? Or does the run request do something with that timestamp data?

I imagine it's the former based on the sentence that follows this one, but being more specific here would be good